docs/path-resolution.rst

```diff
@@ -250,32 +250,23 @@ For example in ``./abc/..//`` there are three path segments: ``.``, ``abc`` and
 
 The compiler computes a source unit name from the import path in the following way:
 
-1. First a prefix is computed
-
-    - Prefix is initialized with the source unit name of the importing source unit.
-    - The last path segment with preceding slashes is removed from the prefix.
-    - Then, the leading part of the normalized import path, consisting only of ``/`` and ``.``
-      characters is considered.
-      For every ``..`` segment found in this part the last path segment with preceding slashes is
-      removed from the prefix.
-
-2. Then the prefix is prepended to the normalized import path.
-   If the prefix is non-empty, a single slash is inserted between it and the import path.
+#. The result is initialized with the source unit name of the importing source unit.
+#. The last path segment with preceding slashes is removed from the result.
+#. Then, for every segment in the import path, starting from the leftmost one:
+    - If the segment is ``.``, it is skipped.
+    - If the segment is ``..``, the last path segment with preceding slashes is removed from the result.
+    - Otherwise, the segment, (preceded by a single slash if the result is not empty), is appended to the result.
```

Comment:
This part was a bit convoluted. By "normalized import path" I meant a path where `.` and `..` segments are resolved as much as possible. I.e. `../a//b/..//c/./d.sol` would be normalized to `../a/c/d.sol`, which has no consecutive slashes and removes the internal `..` and `.` segments but still can have several `..` leading segments because those one can't be matched with anything.

And yeah, it did not mention that these leading `..` segments should be removed before joining the prefix and the normalized path.

I replaced the whole thing with something that I think is simpler and more closely matches the way this is actually implemented: https://github.com/ethereum/solidity/blob/311b2054afae02bfaa8521fd94475a06e0669025/libsolutil/CommonIO.cpp#L141-L159